src/Bridge/Doctrine/MongoDbOdm/CollectionDataProvider.php

```diff
@@ -32,14 +33,16 @@
 final class CollectionDataProvider implements CollectionDataProviderInterface, RestrictedDataProviderInterface
 {
     private $managerRegistry;
+    private $resourceMetadataFactory;
     private $collectionExtensions;
 
     /**
      * @param AggregationCollectionExtensionInterface[] $collectionExtensions
      */
-    public function __construct(ManagerRegistry $managerRegistry, iterable $collectionExtensions = [])
+    public function __construct(ManagerRegistry $managerRegistry, ResourceMetadataFactoryInterface $resourceMetadataFactory, iterable $collectionExtensions = [])
```

Comment:
Could be made optional (for convenience when not using the bundle), and as last parameter (for backward compatibility, even if I know this class is experimental) isn't it?

Reply:
> for convenience when not using the bundle

I don't understand this sentence. If the MongoDB ODM Bundle is not used, this service is not instantiated, so it shouldn't be a problem.

For BC, I don't have a strong opinion but why the need of having the `experimental` annotation then?

Reply:
it is experimental, it could indeed be optional but I'm not sure there's added value to that.